witchcraft-server/src/lib.rs

```diff
@@ -585,15 +584,39 @@ async fn shutdown(shutdown_hooks: ShutdownHooks, timeout: Duration) -> Result<()
     Ok(())
 }
 
-fn signals() -> Result<impl Stream<Item = ()>, Error> {
-    let sigint = signal(SignalKind::interrupt())?;
-    let sigterm = signal(SignalKind::terminate())?;
-    Ok(stream::select(sigint, sigterm))
+#[cfg(unix)]
+mod signals {
+    use conjure_error::Error;
+    use futures_util::{stream, Stream};
+    use tokio::signal::unix::{self, SignalKind};
+
+    pub fn signals() -> Result<impl Stream<Item = ()>, Error> {
+        let sigint = signal(SignalKind::interrupt())?;
+        let sigterm = signal(SignalKind::terminate())?;
+        Ok(stream::select(sigint, sigterm))
+    }
+
+    fn signal(kind: SignalKind) -> Result<impl Stream<Item = ()>, Error> {
+        let mut signal = unix::signal(kind).map_err(Error::internal_safe)?;
+        Ok(stream::poll_fn(move |cx| signal.poll_recv(cx)))
+    }
 }
 
-fn signal(kind: SignalKind) -> Result<impl Stream<Item = ()>, Error> {
-    let mut signal = unix::signal(kind).map_err(Error::internal_safe)?;
-    Ok(stream::poll_fn(move |cx| signal.poll_recv(cx)))
+#[cfg(windows)]
+mod signals {
+    use conjure_error::Error;
+    use futures_util::{stream, Stream};
+    use tokio::signal::windows;
+
+    pub fn signals() -> Result<impl Stream<Item = ()>, Error> {
+        let mut ctrlc = windows::ctrl_c().map_err(Error::internal_safe)?;
+        let ctrlc = stream::poll_fn(move |cx| ctrlc.poll_recv(cx));
+
+        let mut ctrlbreak = windows::ctrl_break().map_err(Error::internal_safe)?;
+        let ctrlbreak = stream::poll_fn(move |cx| ctrlbreak.poll_recv(cx));
```

Comment:
Using ctrl-break here for ETE tests since you apparently can't send ctrl-c to a single process group on windows.